### PR TITLE
Upgrade docker base images

### PR DESCRIPTION
**Pull request by chashikajw**
### Purpose

As $subject, this will upgrade the docker base images for the adapter, enforcer. and router

This will remove vulnerabilities and reduce image sizes as follows.

wso2/mg-router  (114MB ---->  **64.5MB** )   (17.1 ---> 81.3MB)
wso2/mg-adapter  (106MB -----> **38.8MB** )
wso2/mg-enforcer  (remains same --> 214MB)

### Issues
Fixes https://github.com/wso2/product-microgateway/issues/1805

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Ubuntu 20.04

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: router/src/main/resources/Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 # -----------------------------------------------------------------------
 
-FROM envoyproxy/envoy:v1.16.0
+FROM envoyproxy/envoy-alpine:v1.16.2
```
Comment (Rajith90) [inline router/src/main/resources/Dockerfile:17]: Can we upgrade to latest as in https://github.com/wso2/product-microgateway/pull/1786/files, so we can close the 1768 PR and merge this

Comment (chashikajw) [reply router/src/main/resources/Dockerfile]: Upgrade with f42a48093b07ac76356bf50bbbe19cb0cfb3899c

**Comment by VirajSalaka** [comment]
https://github.com/wso2/product-microgateway/pull/1786/files

File: router/src/main/resources/Dockerfile [hunk 2]
```diff
@@ -37,4 +37,4 @@ ENV ENFORCER_CA_CERT_PATH=/home/wso2/security/truststore/mg.pem
...
... TLSv1_2, tls_maximum_protocol_version: TLSv1_2}, tls_certificates: {private_key: {filename: '${ROUTER_PRIVATE_KEY_PATH}'}, certificate_chain: {filename: '${ROUTER_PUBLIC_CERT_PATH}'}}, validation_context: {trusted_ca: {filename: '${ENFORCER_CA_CERT_PATH}'}}}}}, load_assignment: {cluster_name: ext-authz, endpoints: [{lb_endpoints: [{endpoint: {address: {socket_address: {address: '${ENFORCER_HOST}', port_value: '${ENFORCER_PORT}'}}}}]}]}}, {name: token_cluster, type: STRICT_DNS, connect_timeout: 20s, transport_socket: {name: envoy.transport_sockets.tls, typed_config: {'@type': type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext, common_tls_context: {tls_params: {tls_minimum_protocol_version: TLSv1_2, tls_maximum_protocol_version: TLSv1_2}, tls_certificates: {private_key: {filename: '${ROUTER_PRIVATE_KEY_PATH}'}, certificate_chain: {filename: '${ROUTER_PUBLIC_CERT_PATH}'}}, validation_context: {trusted_ca: {filename: '${ENFORCER_CA_CERT_PATH}'}}}}}, load_assignment: {cluster_name: token_cluster, endpoints: [{lb_endpoints: [{endpoint: {address: {socket_address: {address: '${ENFORCER_HOST}', port_value: 8082}}}}]}]}}]}}" --bootstrap-version 3 $TRAILING_ARGS
```
Comment (VirajSalaka) [inline router/src/main/resources/Dockerfile:40]: new line

Comment (chashikajw) [reply router/src/main/resources/Dockerfile]: FIxed in 9e99035a5c437cd13c90ce5815680284bd346b5b

Comment (chashikajw) [reply router/src/main/resources/Dockerfile]: Fixed with c6699d95c6e59fc5c34983823c2c833e84b17ed4

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**